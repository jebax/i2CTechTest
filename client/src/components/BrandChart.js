import React from 'react'
import PropTypes from 'prop-types'
import {
  XYPlot,
  XAxis,
  YAxis,
  VerticalGridLines,
  HorizontalGridLines,
  LineMarkSeries
} from 'react-vis'

const BrandChart = ({ exposed, control }) => {
  return (
    <XYPlot
      height={300}
      width={500}
      xType={'time'}
    >
      <HorizontalGridLines />
      <VerticalGridLines />
      <XAxis />
      <YAxis />
      <LineMarkSeries
        className={'brandExposedData'}
        curve={'curveMonotoneX'}
        data={exposed}
      />
      <LineMarkSeries
        className={'brandControlData'}
        curve={'curveMonotoneX'}
        data={control}
      />
    </XYPlot>
  )
}

BrandChart.propTypes = {
  exposed: PropTypes.array.isRequired,
  control: PropTypes.array.isRequired
}

export default BrandChart
