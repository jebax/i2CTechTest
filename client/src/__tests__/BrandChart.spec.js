import React from 'react'
import { shallow } from 'enzyme'
import BrandChart from '../components/BrandChart'

describe('Brand chart', () => {
  var brandChart
  var exposed
  var control

  beforeAll(() => {
    exposed = [{ x: new Date('2018-01-01'), y: '444' }]
    control = [{ x: new Date('2018-01-01'), y: '333' }]

    brandChart = shallow(
      <BrandChart
        exposed={exposed}
        control={control}
      />
    )
  })

  it('renders correctly', () => {
    expect(brandChart).toMatchSnapshot()
  })

  it('renders a react-vis chart', () => {
    expect(brandChart.find('XYPlot').length).toEqual(1)
  })

  it('displays time on the x-axis', () => {
    const plot = brandChart.find('XYPlot')
    expect(plot.props().xType).toEqual('time')
  })

  it('renders two LineMarkSeries components', () => {
    expect(brandChart.find('LineMarkSeries').length).toEqual(2)
  })

  it('renders the exposed data correctly', () => {
    const firstLine = brandChart.find('LineMarkSeries').first()

    expect(firstLine.props().className).toEqual('brandExposedData')
    expect(firstLine.props().data).toEqual(exposed)
  })

  it('renders the control data correctly', () => {
    const secondLine = brandChart.find('LineMarkSeries').at(1)

    expect(secondLine.props().className).toEqual('brandControlData')
    expect(secondLine.props().data).toEqual(control)
  })
})
