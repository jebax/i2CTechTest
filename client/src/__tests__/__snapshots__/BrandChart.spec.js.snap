// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand chart renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrandChart
    control={
      Array [
        Object {
          "x": 2018-01-01T00:00:00.000Z,
          "y": "333",
        },
      ]
    }
    exposed={
      Array [
        Object {
          "x": 2018-01-01T00:00:00.000Z,
          "y": "444",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <HorizontalGridLines
          attr="y"
          direction="horizontal"
        />,
        <VerticalGridLines
          attr="x"
          direction="vertical"
        />,
        <XAxis
          attr="x"
          attrAxis="y"
          orientation="bottom"
        />,
        <YAxis
          attr="y"
          attrAxis="x"
          orientation="left"
        />,
        <LineMarkSeries
          className="brandExposedData"
          curve="curveMonotoneX"
          data={
            Array [
              Object {
                "x": 2018-01-01T00:00:00.000Z,
                "y": "444",
              },
            ]
          }
          getNull={[Function]}
          lineStyle={Object {}}
          markStyle={Object {}}
          opacity={1}
          stack={false}
          strokeStyle="solid"
          style={Object {}}
        />,
        <LineMarkSeries
          className="brandControlData"
          curve="curveMonotoneX"
          data={
            Array [
              Object {
                "x": 2018-01-01T00:00:00.000Z,
                "y": "333",
              },
            ]
          }
          getNull={[Function]}
          lineStyle={Object {}}
          markStyle={Object {}}
          opacity={1}
          stack={false}
          strokeStyle="solid"
          style={Object {}}
        />,
      ],
      "className": "",
      "height": 300,
      "margin": Object {
        "left": 100,
        "right": 100,
      },
      "width": 500,
      "xType": "time",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "attr": "y",
          "direction": "horizontal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "attr": "x",
          "direction": "vertical",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "attr": "x",
          "attrAxis": "y",
          "orientation": "bottom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "attr": "y",
          "attrAxis": "x",
          "orientation": "left",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "brandExposedData",
          "curve": "curveMonotoneX",
          "data": Array [
            Object {
              "x": 2018-01-01T00:00:00.000Z,
              "y": "444",
            },
          ],
          "getNull": [Function],
          "lineStyle": Object {},
          "markStyle": Object {},
          "opacity": 1,
          "stack": false,
          "strokeStyle": "solid",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "brandControlData",
          "curve": "curveMonotoneX",
          "data": Array [
            Object {
              "x": 2018-01-01T00:00:00.000Z,
              "y": "333",
            },
          ],
          "getNull": [Function],
          "lineStyle": Object {},
          "markStyle": Object {},
          "opacity": 1,
          "stack": false,
          "strokeStyle": "solid",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <HorizontalGridLines
            attr="y"
            direction="horizontal"
          />,
          <VerticalGridLines
            attr="x"
            direction="vertical"
          />,
          <XAxis
            attr="x"
            attrAxis="y"
            orientation="bottom"
          />,
          <YAxis
            attr="y"
            attrAxis="x"
            orientation="left"
          />,
          <LineMarkSeries
            className="brandExposedData"
            curve="curveMonotoneX"
            data={
              Array [
                Object {
                  "x": 2018-01-01T00:00:00.000Z,
                  "y": "444",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />,
          <LineMarkSeries
            className="brandControlData"
            curve="curveMonotoneX"
            data={
              Array [
                Object {
                  "x": 2018-01-01T00:00:00.000Z,
                  "y": "333",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />,
        ],
        "className": "",
        "height": 300,
        "margin": Object {
          "left": 100,
          "right": 100,
        },
        "width": 500,
        "xType": "time",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attr": "y",
            "direction": "horizontal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attr": "x",
            "direction": "vertical",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attr": "x",
            "attrAxis": "y",
            "orientation": "bottom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attr": "y",
            "attrAxis": "x",
            "orientation": "left",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "brandExposedData",
            "curve": "curveMonotoneX",
            "data": Array [
              Object {
                "x": 2018-01-01T00:00:00.000Z,
                "y": "444",
              },
            ],
            "getNull": [Function],
            "lineStyle": Object {},
            "markStyle": Object {},
            "opacity": 1,
            "stack": false,
            "strokeStyle": "solid",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "brandControlData",
            "curve": "curveMonotoneX",
            "data": Array [
              Object {
                "x": 2018-01-01T00:00:00.000Z,
                "y": "333",
              },
            ],
            "getNull": [Function],
            "lineStyle": Object {},
            "markStyle": Object {},
            "opacity": 1,
            "stack": false,
            "strokeStyle": "solid",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
